Agents/FeatureInfoAgent/README.md

```diff
@@ -0,0 +1,94 @@
+# Feature Info Agent
+
+This Feature Info Agent (FIA) acts as a single access point for [DTVF Visualisations](https://github.com/cambridge-cares/TheWorldAvatar/wiki/Digital-Twin-Visualisations) to query for both meta and timeseries data of an individual feature (i.e. a single geographical location) before display within the visualisation's side panel.
+
+## Overview
+
+The FIA is a relatiely simple HTTP Agent built using the JPS Base Lib's agent framework. When a new request is received, the internal process proceeds as follows:
+
+1. Check if the configuration has been loaded.
+   1. If not, then load it.
+   2. Dynamically discover the Ontop endpoint.
+   3. Dynamically discover the PostGreSQL endpoint.
+   4. Dynamically discover all endpoints provided by Blazegraph.
+2. Check the incoming route.
+   1. Throw error if an unknown route.
+3. If `/status` route:
+   1. Return current status of the agent.
+4. If `/get` route:
+   1. Check validity of incoming request.
+      1. If malformed, send a BAD_REQUEST response.
+      2. Determine the class representing the IRI.
+         1. If query successful but no class returned, send a NO_CONTENT response.
+      3. Look up corresponding meta query for that class.
+         1. If none found, send a NO_CONTENT response.
+      4. Run query to get metadata.
+      5. Parse resulting metadata into approved DTVF format.
+      6. Look up corresponding timeseries query for that class.
+         1. If none found, log warning but continue.
+         2. If found, run query.
+            1. If no data found, quietly continue.
+      7. Build and return final JSON object.
+
+It's also worth noting that in the current version of the FIA, any queries to the knowledge graphs are sent to all discovered namespace endpoints via federation. Whilst this will marginally increase processing times, these queries should be pretty quick, and shouldn't be triggered too often, so the risk of delay is hopefully less than the benefit of not having to specify the endpoint beforehand.
```

Comment:
That does sound like a reasonable idea. Would a query to just see if an IRI exists within a 8-10 Mil triplestore be lengthy?

Reply:
It should not. Queries for specific IRIs should be very quick generally; however, depending on the metadata query and in case an IRI exists in multiple namespaces, this could lead to lengthy queries in case the query is federated to the unnecessary endpoint first.

Reply:
Unless we find the current implementation too slow, then I suggest we leave this to be looked into as a new feature within a subsequent version of the agent.